Assets/Scripts/Models/RoomManager.cs

```diff
@@ -0,0 +1,320 @@
+﻿using System;
+using System.Collections.Generic;
+
+public class RoomManager
+{
+    public event Action<Room> Adding;
+
+    public event Action<Room> Added;
+
+    public event Action<Room, Room> Split;
+
+    public event Action<Room, Room> Joined;
+
+    public event Action<Room> Removing;
+
+    public event Action<Room> Removed;
+
+    public List<Room> rooms;
```

Comment:
The big problem is that we can't allow other to add or remove rooms from this list, they need to use the functions. Having it as a public property is not good.

An alternative is having a Count function and a GetRoom function and use a for which should not have any performance problem. It would be cleaner with a GetRooms, but it might be an issue if we need to copy them.